tests/unit/resources/test_resource.py

```diff
@@ -251,6 +250,78 @@ def test_not_wait_for_activity_on_create(self, mock_wait4task, mock_post):
 
         mock_wait4task.assert_not_called()
 
+    @mock.patch.object(connection, 'patch')
+    def test_patch_request_uri_and_body_when_id_is_provided(self, mock_patch):
+        request_body = [{
+            'op': 'replace',
+            'path': '/name',
+            'value': 'new_name',
+        }]
+        mock_patch.return_value = {}, {}
+
+        self.resource_client.patch('123a53cz', 'replace', '/name', 'new_name', False)
+
+        mock_patch.assert_called_once_with('/rest/testuri/123a53cz', request_body)
+
+    @mock.patch.object(connection, 'patch')
+    def test_patch_request_uri_and_body_when_uri_is_provided(self, mock_patch):
```

Comment:
I think this test name is hard to understand. would be better "test_patch_should_use_provided_uri"